pkg/controller/aws/cache/managed.go

```diff
@@ -0,0 +1,255 @@
+	// not. We don't check for the latter here because it's less surprising to
+	// submit the request as the operator intended and let the reconcile fail
+	// with an explanatory message from AWS explaining that transit encryption
+	// is required.
+	if !g.Spec.AuthEnabled {
+		token := ""
+		r := e.client.CreateReplicationGroupRequest(elasticache.NewCreateReplicationGroupInput(g, token))
+		r.SetContext(ctx)
+		_, err := r.Send()
+		return resource.ExternalCreation{}, errors.Wrap(err, errCreateReplicationGroup)
+	}
+
+	token, err := util.GeneratePassword(maxAuthTokenData)
+	if err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, errGenerateAuthToken)
+	}
+
+	r := e.client.CreateReplicationGroupRequest(elasticache.NewCreateReplicationGroupInput(g, token))
+	r.SetContext(ctx)
+	if _, err := r.Send(); err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, errCreateReplicationGroup)
+	}
+
+	c := resource.ExternalCreation{
+		ConnectionDetails: resource.ConnectionDetails{
+			corev1alpha1.ResourceCredentialsSecretPasswordKey: []byte(token),
+		},
+	}
+
+	return c, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (resource.ExternalUpdate, error) {
+	g, ok := mg.(*v1alpha1.ReplicationGroup)
+	if !ok {
+		return resource.ExternalUpdate{}, errors.New(errNotReplicationGroup)
+	}
+
+	dr := e.client.DescribeReplicationGroupsRequest(elasticache.NewDescribeReplicationGroupsInput(g))
+	dr.SetContext(ctx)
+	rsp, err := dr.Send()
+	if err != nil {
+		return resource.ExternalUpdate{}, errors.Wrap(err, errDescribeReplicationGroup)
+	}
+
+	// DescribeReplicationGroups can return one or many replication groups. We
+	// ask for one group by name, so we should get either a single element list
+	//  or an error.
+	if elasticache.ReplicationGroupNeedsUpdate(g, rsp.ReplicationGroups[0]) {
+		mr := e.client.ModifyReplicationGroupRequest(elasticache.NewModifyReplicationGroupInput(g))
+		mr.SetContext(ctx)
+		_, err = mr.Send()
+		return resource.ExternalUpdate{}, errors.Wrap(err, errModifyReplicationGroup)
+	}
+
+	for _, cc := range g.Status.MemberClusters {
+		dcc := e.client.DescribeCacheClustersRequest(elasticache.NewDescribeCacheClustersInput(cc))
+		dcc.SetContext(ctx)
+		rsp, err := dcc.Send()
+		if err != nil {
+			return resource.ExternalUpdate{}, errors.Wrapf(err, errDescribeCacheCluster)
+		}
+
+		// DescribeCacheClusters can return one or many cache clusters. We ask
+		// for one cluster by name, so we should get either a single element
+		// list or an error.
+		if elasticache.CacheClusterNeedsUpdate(g, rsp.CacheClusters[0]) {
+			mr := e.client.ModifyReplicationGroupRequest(elasticache.NewModifyReplicationGroupInput(g))
+			mr.SetContext(ctx)
+			_, err = mr.Send()
+			return resource.ExternalUpdate{}, errors.Wrap(err, errModifyReplicationGroup)
+		}
+	}
+
+	return resource.ExternalUpdate{}, nil
+}
+
+func (e *external) Finalize(ctx context.Context, mg resource.Managed) error {
```

Comment:
Observation here that reclamation behavior is still left to the designer of the controller. Some thoughts on `reclaimPolicy` have been captured in [here](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-resource-reclaim-policy.md). I don't think that we will be able to dictate a single behavior for all types of resources, but would be interesting to explore a default behavior.

Reply:
Right. I've called the `Finalize` (not `Delete`) since it does not necessarily delete the external resource, unless asked to do so by the reclaim policy. We may consider moving the reclaim logic into this reconciler once we move to the pattern your design proposes.  I'd also be fine to leave this up to the `Finalize` implementation.

Reply:
We ended up discussing this topic in more detail in https://github.com/crossplaneio/crossplane/pull/593#discussion_r305821939, so I'll resolve this thread.